lib/services/identitycenter.go

```diff
@@ -226,3 +236,84 @@ type IdentityCenter interface {
 	IdentityCenterPrincipalAssignments
 	IdentityCenterAccountAssignments
 }
+
+// NewIdentityCenterAccountMatcher creates a new [IdentityCenterMatcher]
+// configured to match the supplied [IdentityCenterAccount].
+func NewIdentityCenterAccountMatcher(account IdentityCenterAccount) *IdentityCenterMatcher {
+	return &IdentityCenterMatcher{
+		accountID:        account.GetSpec().GetId(),
+		permissionSetARN: nil,
+	}
+}
+
+// NewIdentityCenterAccountAssignmentMatcher creates a new [IdentityCenterMatcher]
+// configured to match the supplied [IdentityCenterAccountAssignment].
+func NewIdentityCenterAccountAssignmentMatcher(account IdentityCenterAccountAssignment) *IdentityCenterMatcher {
+	psARN := account.GetSpec().GetPermissionSet().GetArn()
+	return &IdentityCenterMatcher{
+		accountID:        account.GetSpec().GetAccountId(),
+		permissionSetARN: &psARN,
+	}
+}
+
+// IdentityCenterMatcher implements a [RoleMatcher] for comparing Identity Center
+// resources against the AccountAssignments specified in a Role condition.
+//
+// The same type is used for matching both [IdentityCenterAccount]s and
+// [IdentityCenterAccountAssignment]s, the permission set is `nil` when matching
+// an Account.
+type IdentityCenterMatcher struct {
+	accountID        string
+	permissionSetARN *string
+}
+
+// Match implements RoleMatcher for IdentityCenterMatcher. It attempts to match
+// the Account Assignments in a Role Condition against a known Account ID or
+// (Account ID, PermissionSetARN) pair
+func (m *IdentityCenterMatcher) Match(role types.Role, condition types.RoleConditionType) (bool, error) {
+	for _, asmt := range role.GetIdentityCenterAccountAssignments(condition) {
+		accountMatches, err := m.matchExpression(m.accountID, asmt.Account)
+		if err != nil {
+			return false, trace.Wrap(err)
+		}
+
+		if !accountMatches {
+			continue
+		}
+
+		if m.permissionSetARN == nil {
+			return true, nil
+		}
+
+		psMatches, err := m.matchExpression(*(m.permissionSetARN), asmt.PermissionSet)
+		if err != nil {
+			return false, trace.Wrap(err)
+		}
+
+		if psMatches {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func (m *IdentityCenterMatcher) matchExpression(target, accountExpression string) (bool, error) {
```

Comment:
Nit: This is used to match both accounts and permission sets from what I can see.
```suggestion
func (m *IdentityCenterMatcher) matchExpression(target, expression string) (bool, error) {
```